### ISSUE TITLE
Problems with reproducing ADP and Diffpure

### ISSUE THREAD

**Issue by phoshowy** [ml-postech/robust-evaluation-of-diffusion-based-purification#4]
Hi! I'm a freshman in AP, and I'm recently trying to reproduce the experimental results in your article, Table2, and have some uncertainties about it eager to get your help!
First of all, after cloning your project,  I successfully ran **cifar10_pgd.sh**, will I be able to get the results in Table2 regarding Diffpure (PGD+EOF), and if I need to change the content of cifar10_pgd.sh, cifar10.yml, or any other files, please let me know!
Another issue is that there are some modules that don't seem to be found in your project (maybe I'm careless), such as AutoAttack and ADP, I've cloned AA's repository and added to the original code according to the guidelines in your article, but I still haven't found anything related to running ADP, please guide me!
Sorry to bother you and thanks for your help!

**Comment by Minjong-Lee** [comment]
Thank you for your interest in our work.

Unfortunately, we currently don't provide code for ADP and DiffPure in this repository. The script you ran, cifar10_pgd.sh, is also technically different from DiffPure (PGD+EOF). DiffPure uses an SDE/ODE solver, while our repository only provides attacks and defenses based on DDPM/DDIM. In my experience, this difference significantly impacts the results.

We used the original code from each method for the evaluations. If you want to replicate experiments similar to those described in our paper for ADP and DiffPure, you will need to modify and run the code from their original repositories. If I remember correctly, modifying both methods are relatively straightforward.

For example, in the case of DiffPure, [this part of code](https://github.com/NVlabs/DiffPure/blob/985703facc13da38d84bcaf52b3bc5b2ef9906f1/runners/diffpure_ode.py#L229) calculates gradients using the adjoint method. By commenting this out, you can perform gradient calculations through backpropagation, as we did in our paper. (However, keep in mind that for the surrogate process, the denoising processes used for attacks and defenses are different.) Similarly, for ADP, attacks are defined at [here](https://github.com/jmyoon1/adp/tree/master/attacks), where you can configure the gradient computation for the full process. Setting this up should be simpler than you might expect.

When organizing this code, I didn't significantly modify the original implementations, so I didn't separately document each method's code. I apologize if this made reproducing the experiments more difficult. If you need further, feel free to ask.

**Comment by phoshowy** [comment]
Thank you for your reply, I am experimenting with your guidelines and hope to get good results!

**Comment by phoshowy** [comment]
I'm sorry to bother you again for some questions.

In the reproduction of the original diffpure,  is this experimental configuration the same as in the original script?

Also, and perhaps a less appropriate question, there is a file that appears in the ADP [code](https://github.com/jmyoon1/adp/blob/master/runners/empirical.py#L88) that I can't get, and I can't find an answer to that question in the issue section of the original repository, so do you know where to get that file?

**Comment by Minjong-Lee** [comment]
Sorry for the late reply.

> In the reproduction of the original diffpure, is this experimental configuration the same as in the original script?

Yes. Except for the hyperparameters mentioned in our paper (e.g., t*), everything else is the same as the original.

> Also, and perhaps a less appropriate question, there is a file that appears in the ADP [code](https://github.com/jmyoon1/adp/blob/master/runners/empirical.py#L88) that I can't get, and I can't find an answer to that question in the issue section of the original repository, so do you know where to get that file?

To be honest, I don’t remember that part of the code clearly. I vaguely recall that the weights for that model are available online, but I’m not sure. Sorry I couldn’t be more helpful. In our paper, we only use the classifiers provided by RobustBench. For the standard WideResNet-28-10, we load it as follows:
`
from robustbench import load_model;
load_model(model_name='Standard', dataset='cifar10', threat_model='Linf')
`

**Comment by phoshowy** [comment]
Thank you for your kind guidance！
